src/vm/wren_compiler.c

```diff
@@ -3202,6 +3206,21 @@ static bool method(Compiler* compiler, Variable classVariable)
     createConstructor(compiler, &signature, methodSymbol);
     defineMethod(compiler, classVariable, true, constructorSymbol);
   }
+  
+  // Check for duplicate methods. Doesn't matter that it's already been
+  // defined, error will discard bytecode anyway.
+  IntBuffer* methods = isStatic ? &compiler->enclosingClass->staticMethods
+                                : &compiler->enclosingClass->instanceMethods;
+  
+  int symbol = wrenIntBufferFind(methods, methodSymbol);
+  if (symbol != -1)
+  {
+    const char* methodType = isStatic ? "static method" : "method";
+    error(compiler, "Already defined %s '%s' for class %s in module '%s'.",
+      methodType, fullSignature, &compiler->enclosingClass->name->value,
+      compiler->parser->module->name);
```

Comment:
Yeah, runtime errors are a little different. :) Compile errors handle it automatically because at compile time we know the most recently consumed token and can infer source position based on that.